### PR TITLE
Add live mode

### PR DESCRIPTION
**Pull request by juliangruber**
the most important parts to look at:

- the api: https://github.com/juliangruber/hyperdrive-import-files/blob/3d002979fa149e50c8e711496fe32ec7b69c27e8/README.md#hyperimportarchive-files--options-cb
- the new implementation, which in general has been refactored quite a bit: https://github.com/juliangruber/hyperdrive-import-files/blob/3d002979fa149e50c8e711496fe32ec7b69c27e8/index.js

cc @mafintosh

### CODE REVIEW

**Comment by juliangruber** [comment]
this is required for https://github.com/juliangruber/dat-desktop/issues/27

File: test/index.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ const hyperdrive = require('hyperdrive')
 const memdb = require('memdb')
 const hyperImport = require('..')
 
+const sort = entries => entries.sort((a, b) => b.name - a.name)
```
Comment (mafintosh) [inline test/index.js]: does this work? i usually to `b.name.localeCompare(a.name)`

Comment (juliangruber) [inline test/index.js]: i guess it's less correct but yeah it's def. good enough here

Comment (juliangruber) [inline test/index.js]: i thought it worked, but fixed it now to do the localeCompare

**Comment by mafintosh** [comment]
looks good!

**Comment by karissa** [comment]
should we use this in maxogden/dat? ping @jhand

**Comment by joehand** [comment]
Ya, that'd be great! There needs to be a few more things so we can use it in the CLI:

* Resume import: only add new files from `dir` to the archive.
* Progress/watching events (may be able to do this if we add append events to hyperdrive).

**Comment by juliangruber** [comment]
@joehand sounds good, let's add those next!

**Comment by juliangruber** [comment]
published as 1.2.0

**Comment by juliangruber** [comment]
see

https://github.com/juliangruber/hyperdrive-import-files/issues/3

https://github.com/juliangruber/hyperdrive-import-files/issues/2

**Pull request merged**